src/components/Tool/FixPhoneNumbers/FixPhoneNumbers.tsx

```diff
@@ -232,100 +196,100 @@ const FixPhoneNumbers: React.FC = () => {
-                        handleSourceChange(event)
-                      }
-                    >
-                      <option value="MPDX">MPDX</option>
-                      <option value="DataServer">DataServer</option>
-                    </NativeSelect>
-                    <Button className={classes.buttonBlue}>
-                      <Icon
-                        path={mdiCheckboxMarkedCircle}
-                        size={0.8}
-                        className={classes.buttonIcon}
+                      <NativeSelect
+                        input={<StyledInput />}
+                        className={classes.nativeSelect}
+                        value={defaultSource}
+                        data-testid="source-select"
+                        onChange={(
+                          event: React.ChangeEvent<HTMLSelectElement>,
+                        ) => handleSourceChange(event)}
+                      >
+                        <option value="MPDX" data-testid="source-option-mpdx">
+                          MPDX
+                        </option>
+                        <option
+                          value="DataServer"
+                          data-testid="source-option-dataserver"
+                        >
+                          DataServer
+                        </option>
+                      </NativeSelect>
+                      <Button
+                        className={classes.buttonBlue}
+                        data-testid="source-button"
+                      >
+                        <Icon
+                          path={mdiCheckboxMarkedCircle}
+                          size={0.8}
+                          className={classes.buttonIcon}
+                        />
+                        {t('Confirm {{amount}} as {{source}}', {
+                          amount: dataState.length,
+                          source: defaultSource,
+                        })}
+                      </Button>
+                    </Box>
+                  </Box>
+                </Grid>
+                <Grid item xs={12}>
+                  {dataState.map((person, index) => (
+                    <Contact
+                      name={`${person.firstName} ${person.lastName}`}
+                      key={person.id}
+                      contactIndex={index}
+                      numbers={person.phoneNumbers.nodes}
+                      handleChange={handleChange}
+                      handleDelete={handleDeleteModalOpen}
+                      handleAdd={handleAdd}
+                      handleChangePrimary={handleChangePrimary}
+                    />
+                  ))}
+                </Grid>
+                <Grid item xs={12}>
+                  <Box className={classes.footer}>
+                    <Typography>
+                      <Trans
+                        defaults="Showing <bold>{{value}}</bold> of <bold>{{value}}</bold>"
+                        values={{ value: dataState.length }}
```

Comment:
I know this is still being worked on but was curious, this will need to be two numbers right? One for the current length of ```dataState``` and then eventually the complete total number of invalid phone numbers. Also will this page be infinite scroll? I'm not sure the average user would have more than 25 invalid phone numbers but you never know.

Reply:
Yeah this will be two numbers when the pagination/scrolling gets implemented